### PR TITLE
fix bug that prevents single phase data pipelines

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

File: cdap-app-templates/cdap-etl/cdap-etl-data-pipeline/src/main/java/co/cask/cdap/etl/datapipeline/SmartWorkflow.java [hunk 1]
```diff
@@ -76,7 +74,26 @@ protected void configure() {
...
 
+    // if there are no connections, we have no dag
+    if (plan.getPhaseConnections().isEmpty()) {
+      // single phase, just add the program directly
+      if (plan.getPhases().size() == 1) {
+        addProgram(plan.getPhases().keySet().iterator().next(), new TrunkProgramAdder(getConfigurer()));
+        return;
+      }
+
+      // multiple phases, do a fork then join
```
Comment (anwar6953) [inline cdap-app-templates/cdap-etl/cdap-etl-data-pipeline/src/main/java/co/cask/cdap/etl/datapipeline/SmartWorkflow.java]: there can be multiple phases, but no connections between them?

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-DUT3770

File: cdap-app-templates/cdap-etl/cdap-etl-data-pipeline/src/main/java/co/cask/cdap/etl/datapipeline/SmartWorkflow.java (hunk 1 above)
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-data-pipeline/src/main/java/co/cask/cdap/etl/datapipeline/SmartWorkflow.java]: yep, like in the simple multi source test case

Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-etl-data-pipeline/src/main/java/co/cask/cdap/etl/datapipeline/SmartWorkflow.java]: hm actually thats a bug...

**Comment by anwar6953** [comment]
LGTM

**Pull request merged**